rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/ReSubmitJobCommand.java

```diff
@@ -0,0 +1,169 @@
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URLConnection;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.ow2.proactive_grid_cloud_portal.cli.ApplicationContext;
+import org.ow2.proactive_grid_cloud_portal.cli.CLIException;
+import org.ow2.proactive_grid_cloud_portal.cli.cmd.AbstractCommand;
+import org.ow2.proactive_grid_cloud_portal.cli.cmd.Command;
+import org.ow2.proactive_grid_cloud_portal.scheduler.dto.JobIdData;
+import org.ow2.proactive_grid_cloud_portal.scheduler.exception.JobCreationRestException;
+
+
+public class ReSubmitJobCommand extends AbstractCommand implements Command {
+    private final String jobId;
+
+    private String variables;
+
+    private static final Logger logger = null;
+
+    private final JobKeyValueTransformer jobKeyValueTransformer;
+
+    public ReSubmitJobCommand(String... params) throws NullPointerException {
+        Objects.requireNonNull(params);
+
+        this.jobId = params[0];
+        if (params.length > 1) {
+            this.variables = params[1];
+        }
+        this.jobKeyValueTransformer = new JobKeyValueTransformer();
+
+    }
+
+    @Override
+    public void execute(ApplicationContext currentContext) throws CLIException {
+
+        try {
+            validateFilePath(currentContext);
+            File jobFile = new File(pathname);
+            String contentType = URLConnection.getFileNameMap().getContentTypeFor(pathname);
+            JobIdData jobId;
+            try (FileInputStream inputStream = new FileInputStream(jobFile)) {
+                if (APPLICATION_XML.getMimeType().equals(contentType)) {
+                    jobId = currentContext.getRestClient().submitXml(currentContext.getSessionId(),
+                                                                     inputStream,
+                                                                     map(this.variables));
+                } else {
+                    jobId = currentContext.getRestClient().submitJobArchive(currentContext.getSessionId(),
+                                                                            inputStream,
+                                                                            map(this.variables));
+                }
+            }
+            writeLine(currentContext, "Job('%s') successfully submitted: job('%d')", pathname, jobId.getId());
+            resultStack(currentContext).push(jobId);
+        } catch (Exception e) {
+            handleError(String.format("An error occurred while attempting to submit job('%s'):", pathname),
+                        e,
+                        currentContext);
+        }
+
+    }
+
+    private void validateFilePath(ApplicationContext currentContext) throws JobCreationRestException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "currentContext". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)